lib/os/mem.c

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright (c) 2021 Carlo Caione <ccaione@baylibre.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <stddef.h>
+#include <zephyr/types.h>
+#include <errno.h>
+#include <sys/util.h>
+
+void *sys_memset(void *s, int c, size_t count)
+{
+	unsigned long *sl = (unsigned long *) s;
+	unsigned long cl = 0;
+	char *s8;
+	int i;
+
+	if (((unsigned long)s & (sizeof(*sl) - 1)) == 0) {
```

Comment:
If the request is to set a large region where the start address isn't word-aligned the behavior is pessimized.  Is it worth adding a pre-align loop like the post-align loop to deal with this?

Reply:
We can use whatever memset() we want, I just took one from the linux kernel without much thought TBH. In theory I could have copied the one shipped in the minimal libc that we already ship in `lib/libc/minimal/source/string/string.c`. What about that one?

Reply:
Moving that implementation here, and just calling it from minimal libc, would probably be better.  I'm now concerned about "just took the one from the linux kernel", since that's got licensing/copyright implications.

Reply:
I know, but there are only a few permutations of a memset() around and in general this is an RFC because I was more concerned with the approach to fix the issue than the actual code :) I'll move that implementation here, no prob.